token-group/example/tests/setup.rs

```diff
@@ -0,0 +1,81 @@
+#![cfg(feature = "test-sbf")]
+
+use {
+    solana_program_test::{processor, tokio::sync::Mutex, ProgramTest, ProgramTestContext},
+    solana_sdk::{pubkey::Pubkey, signature::Keypair, signer::Signer},
+    spl_token_client::{
+        client::{
+            ProgramBanksClient, ProgramBanksClientProcessTransaction, ProgramClient,
+            SendTransaction, SimulateTransaction,
+        },
+        token::{ExtensionInitializationParams, Token},
+    },
+    spl_token_metadata_interface::state::TokenMetadata,
+    std::sync::Arc,
+};
+
+/// Set up a program test
+pub async fn setup_program_test(
+    program_id: &Pubkey,
+) -> (
+    Arc<Mutex<ProgramTestContext>>,
+    Arc<dyn ProgramClient<ProgramBanksClientProcessTransaction>>,
+    Arc<Keypair>,
+) {
+    let mut program_test = ProgramTest::new(
+        "spl_token_group_example",
+        *program_id,
+        processor!(spl_token_group_example::processor::process),
+    );
+    program_test.prefer_bpf(false);
+    program_test.add_program(
+        "spl_token_2022",
+        spl_token_2022::id(),
+        processor!(spl_token_2022::processor::Processor::process),
+    );
+    let context = program_test.start_with_context().await;
+    let payer = Arc::new(context.payer.insecure_clone());
+    let context = Arc::new(Mutex::new(context));
+    let client: Arc<dyn ProgramClient<ProgramBanksClientProcessTransaction>> =
+        Arc::new(ProgramBanksClient::new_from_context(
+            Arc::clone(&context),
+            ProgramBanksClientProcessTransaction,
+        ));
+    (context, client, payer)
+}
+
+/// Set up a Token-2022 mint and metadata
+pub async fn setup_mint_and_metadata<T: SendTransaction + SimulateTransaction>(
+    token_client: &Token<T>,
+    mint_keypair: &Keypair,
+    mint_authority_keypair: &Keypair,
+    metadata_pubkey: &Pubkey,
+    metadata_update_authority_pubkey: &Pubkey,
+    token_metadata: &TokenMetadata,
+    payer: Arc<Keypair>,
+) {
+    token_client
+        .create_mint(
+            &mint_authority_keypair.pubkey(),
+            None,
+            vec![ExtensionInitializationParams::MetadataPointer {
+                authority: Some(*metadata_update_authority_pubkey),
+                metadata_address: Some(*metadata_pubkey),
+            }],
+            &[mint_keypair],
+        )
+        .await
+        .unwrap();
+    token_client
+        .token_metadata_initialize_with_rent_transfer(
+            &payer.pubkey(),
+            metadata_update_authority_pubkey,
+            &mint_authority_keypair.pubkey(),
+            token_metadata.name.clone(),
+            token_metadata.symbol.clone(),
+            token_metadata.uri.clone(),
+            &[&payer, mint_authority_keypair],
+        )
+        .await
+        .unwrap();
+}
```

Comment:
This is kinda doing something weird, because it's initializing the metadata pointer to point at `metadata_pubkey` but then initializing the metadata on the mint directly, which means that there's never any interaction between the metadata and the collection. Maybe this can simply remove all the metadata bits?

Reply:
sure, makes sense. I've removed metadata from the collections example.